projects/developer/src/app/dashboard/component.ts

```diff
@@ -106,10 +106,13 @@ export class DashboardComponent implements OnInit, OnDestroy {
         }
     }
     ngOnInit() {
-        this.refreshAllJobTypes();
-        this.jobsService.favoritesUpdated.subscribe(() => {
+        if (this.favoriteJobTypes) {
+            this.jobsService.favoritesUpdated.subscribe(() => {

```

Comment:
Could you also unsubscribe to this? I'm wondering if navigating back and forth to the dashboard is also holding onto these API calls. A pattern I usually do:
- change `this.subscription` into `this.subscriptions = []`
- in `this.unsubscribe()` loop over it, `this.subscriptions.forEach(s => s.unsubscribe()); this.subscriptions = [];`
- anywhere that's a subscription push it into the array, `this.subscriptions.push( this.jobsService.favoritesUpdated.subscribe( ... ) );`

Reply:
Ok I was able to fix that as well